### PR TITLE
feat: Increase color palette size, revamp appearance settings page

### PR DESCRIPTION
**Pull request by filp**
Includes commits from https://github.com/lightdash/lightdash/pull/8870 - will rebase once that is merged.

### Description:

- Sets color palette size to 20, and revamps the appearance settings page:

image

- Adds a per-color-input restore option when a value is changed, which restores to either the previous saved value, or a color theme's default

image

- Adds a set of chart color themes, and an interaction to toggle between them:

image

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: packages/frontend/src/components/UserSettings/AppearanceSettingsPanel/index.tsx [hunk 1]
```diff
@@ -0,0 +1,503 @@
...
+import MantineIcon from '../../common/MantineIcon';
+import { SettingsCard } from '../../common/Settings/SettingsCard';
+
+interface ColorPalette {
+    name: string;
+    colors: string[];
+    icon: typeof IconBolt;
+}
+
+const defaultColorPalettes: ColorPalette[] = [
```
Comment (rephus) [inline packages/frontend/src/components/UserSettings/AppearanceSettingsPanel/index.tsx:45]: we could move this to a separate file, even json. But we can keep it here for now

**Review by rephus — APPROVED** [review]
Code looks good. I can't test this on render because this is a sub-branch. If you merge this into another branch, I cant test it there

**Comment by hamzahc1** [comment]
@filp - I wanted to check some behaviour here. Currently, if I update the colour palette in Lightdash, existing charts get the updated colour palette (where I haven't manually changed the colours on a chart) - is this behaviour still there with this PR?

**Comment by filp** [comment]
This is currently pending wrapping up https://github.com/lightdash/lightdash/pull/8870

**Pull request merged**